scripts/core/editor3/actions/spellchecker.jsx

```diff
@@ -21,14 +16,61 @@ export function replaceWord(word) {
 
 /**
  * @ngdoc method
- * @name Spellchecker Actions#showContextMenu
- * @param {Object} contextMenuData
+ * @name replaceWord
+ * @param {String} word
+ * @return {String} action
+ * @description Creates the add word action
+ */
+export function addWord(word) {
+    const spellcheck = ng.get('spellcheck');
+
+    spellcheck.addWord(word, false);
+
+    return replaceWord(word);
+}
+
+/**
+ * @ngdoc method
+ * @name replaceWord
+ * @param {String} word
+ * @return {String} action
+ * @description Creates the ignore word action
+ */
+export function ignoreWord(word) {
+    const spellcheck = ng.get('spellcheck');
+
+    spellcheck.addWord(word, true);
+
+    return replaceWord(word);
```

Comment:
I see. here it could use list of ignored words in state, but the issue is still the same in `addWord` where you add a new word to spellchecker dictionary. probably the redux way would be to keep dictionaries for spellchecking in state